Localization: translation key exceeds character limit for GlotPress

mindgraffiti opened the pull request:
Fixes #15353 

### To test
It can't really be tested until the new translation strings are available. So the only thing to look for is: ensure the localized strings do not exceed 255 characters in English.

1. Change device language to a language that uses words that have more characters than English, such as German or Dutch. 
2. Navigate to My Site > Settings > Start Over. The detailed instructions text should now to be translated instead of falling back to English.

### Screenshots
Proof of bug:
<a href="https://user-images.githubusercontent.com/1062444/99831606-f38f3d80-2b24-11eb-8d14-6d202045ee59.png"><img src="https://user-images.githubusercontent.com/1062444/99831606-f38f3d80-2b24-11eb-8d14-6d202045ee59.png" width="350" /></a>

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

mindgraffiti committed:
Format detailed text so it does not break GlotPress char limits

Limited to 256 chars

mindgraffiti committed:
Remove contraction

Per App's writing guidelines, begin removing contractions wherever they are found in user-facing strings

mindgraffiti committed:
Add release note

AliSoftware commented:
@mindgraffiti Thanks or the change; did you also made the change in GlotPress? 🤔 

💭 If we let GlotPress in its current state on the next iteration our process will create 2 new keys (for the ones you created for paragraphs 1 and 2) that won't be translated in other languages at first, despite us already having the translation of those 2 paragraphs in the old key that we'll soon delete. This means that our contributors will see 2 new keys not being translated yet and will translate them, potentially with slight differences compared to the current translation we use. Not sure if that's a big deal, but I was thinking reporting those translations ourselves in GlotPress manually once the 2 new keys are created might avoid this gotcha?

mindgraffiti commented:
@AliSoftware nope. After the localized strings are created in the class, Platform 9 usually takes over and "magic happens" 😀 . If you'll explain how I should run the script that gathers all the localized strings, I can make an attempt.

AliSoftware commented:
yeah actually while I was writing the comment above that let me thinking, I'll need to investigate a bit deeper what's the best course of action from here:

 - merge the PR without changing GlotPress, and wait for the next code freeze to let the magic happen (that's when all the localizations are merged into a file that GlotPress will pick up / pull regularly); at which point we'd have the new keys in GlotPress and we could finally do the copy/pasting of those translations for all locales manually before any contributor steps in to provide a slightly different one
 - or we trigger the magic sooner than the code freeze, forcing it to happen on demand, to be able to fix it right away. But I'm realizing that it could have unintended consequences, like uploading strings that are in develop while we're not yet sure they'll stay or not change until the next code freeze, so contributors might see new strings to translate before they're supposed to, etc;

It's Friday night already so I'll wait for next week to get some good rest and thing this through, get a grasp at the potential consequences and see what's best… but I'm starting to think option 1 (don't touch GlotPress until next code freeze) might be the best option after all.

ScoutHarris commented on WordPress/Classes/ViewRelated/Blog/Site Management/StartOverViewController.swift:
```diff
@@ -15,7 +15,15 @@ open class StartOverViewController: UITableViewController, MFMailComposeViewCont
     @objc let headerView: TableViewHeaderDetailView = {
         let header = NSLocalizedString("Let Us Help", comment: "Heading for instructions on Start Over settings page")
-        let detail = NSLocalizedString("If you want a site but don't want any of the posts and pages you have now, our support team can delete your posts, pages, media, and comments for you.\n\nThis will keep your site and URL active, but give you a fresh start on your content creation. Just contact us to have your current content cleared out.", comment: "Detail for instructions on Start Over set ...
+        let detail1 = NSLocalizedString("If you want a site but do not want any of the posts and pages you have now, " +
```
I'm totally lurking here, but I have one request. Can you add a comment explaining why this is split? So the next person who comes along doesn't say "oh this can be one string" and jam it all together again.

mindgraffiti replied:
good point! I'll do that.

AliSoftware replied:
Took the liberty to update the comment, as the 255-chars restriction is only on the key, not the translations.

Since it happens that on iOS we use the English copy as the key, then yes we will need to split that key which ends up splitting the English copy. But the translations for that key can be longer.

mindgraffiti committed:
Add a comment explaining why we are breaking apart the strings

mindgraffiti committed:
Add the github issue as well

AliSoftware committed:
Update comment

Only the *key* is limited to 255 characters. The translations (English as well as other locales) are not limited in length

AliSoftware reviewed (approved):
I'll merge this without waiting for a change in GlotPress after all.

Since on next Monday we will start a new Sprint and I'll do a code freeze, which is what initiates an update of the `Localizable.strings` from scanning `NSLocalizedString(…)` calls in our code, then let GlotPress import that updated `.strings` file, the new key will start being available in GP at that point anyway.
